src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,139 +53,175 @@ export interface MenuProps {
 });
-const menuItemAttrs = ({
-  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
-  name, label, ...itemAttrs
-}: MenuRenderItem) => itemAttrs;
-const itemsToRender = computed<MenuRenderItem[]>(() => (props.items.map((item, key) => {
+provide('enableKeyboardNavigation', props.enableKeyboardNavigation);
+const itemsToRender = computed<MenuRenderItem[]>(() => (props.items.map((item, index) => {
   if (typeof item === 'string') {
     return {
-      name: `menu-item-${key}`,
+      name: `menu-item-${index}`,
       label: item,
     };
   }
   return {
-    name: `menu-item-${key}`,
+    name: `menu-item-${index}`,
     ...item,
   };
 })));
-const menuItems = ref<MenuItem[] | []>([]);
-provide('menuItems', menuItems);
-const mItems = computed(() => (menuItems.value));
+const internalMenuItemsTemplateRefs = ref<InstanceType<typeof UiMenuItem>[]>([]);
+const usedMenuTemplateRefs = computed(() => (props.menuItemsTemplateRefs || internalMenuItemsTemplateRefs.value));
 const {
-  firstMenuItem,
-  lastMenuItem,
-  nextMenuItem,
-  prevMenuItem,
-  selectedMenuItem,
-  focusedMenuItem,
-} = useMenuItems(mItems);
-const handleMenuKeydown = async ({ key }: KeyboardEvent) => {
-  if (!props.enableKeyboardNavigation) return;
+  focusedElement,
+  initialElement,
+  selectedElement,
+  nextElement,
+  prevElement,
+} = useArrowNavigation(usedMenuTemplateRefs);
+const setNegativeTabindexForNonInitialMenuItems = async () => {
+  usedMenuTemplateRefs.value.forEach((item) => {
+    if (item === initialElement.value) {
+      return;
+    }
+    // eslint-disable-next-line no-param-reassign
+    item.tabindex = -1;
+  });
+};
+watch(() => (props.enableKeyboardNavigation), async () => {
+  await nextTick();
+  if (props.enableKeyboardNavigation
+      && usedMenuTemplateRefs.value.length > 0) {
+    setNegativeTabindexForNonInitialMenuItems();
+  } else {
+    usedMenuTemplateRefs.value.forEach((item) => {
+      // eslint-disable-next-line no-param-reassign
+      item.tabindex = 0;
+    });
+  }
+});
+onMounted(async () => {
+  await nextTick();
+  if (props.enableKeyboardNavigation
+      && usedMenuTemplateRefs.value.length > 0) {
+    setNegativeTabindexForNonInitialMenuItems();
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    if (props.enableKeyboardNavigation
+        && usedMenuTemplateRefs.value.length < 1
+    ) {
+      console.warn('@infermedica/component-library: use itemsTemplateRefs to pass UiMenuItems template refs.');
+    }
+  }
+});
+const lastFocusedMenuItemTemplateRefs = ref<ElementRef | null>(null);
+const handleMenuItemFocus = async (element: ElementRef | InstanceType<typeof UiMenuItem> | undefined) => {
+  if (element && initialElement.value) {
+    await focusElement(element?.itemTemplateRefs?.content?.$el, true);
+    initialElement.value.tabindex = -1;
+  }
+};
+const handleMenuKeyDown = async ({ key }: KeyboardEvent) => {
+  if (!props.enableKeyboardNavigation) {
+    return;
+  }
```

Comment:
Why not like that? Doesn't eslint allow that?
```suggestion
  if (!props.enableKeyboardNavigation) return;
```